src/core/TSDB.java

```diff
@@ -988,12 +988,12 @@ public WritableDataPoints newBatch(String metric, Map<String, String> tags) {
         Deferred<Object> result = null;
         if (config.enable_appends()) {
           final AppendDataPoints kv = new AppendDataPoints(qualifier, value);
-          final AppendRequest point = new AppendRequest(table, row, FAMILY, 
-              AppendDataPoints.APPEND_COLUMN_QUALIFIER, kv.getBytes());
+          final AppendRequest point = RequestBuilder.buildAppendRequest(config, table, row, FAMILY, 
+              AppendDataPoints.APPEND_COLUMN_QUALIFIER, kv.getBytes(), timestamp);
```

Comment:
This is handled by RequestBuilder.java, which checks the unit of timestamp via masking it and then issuing appropriate Request. If the given timestamp is in seconds. it is multiplied by 1000 to keep HBase happy!